dnf5/commands/download/download.cpp

```diff
@@ -56,13 +78,19 @@ void DownloadCommand::configure() {
     }
 
     context.update_repo_load_flags_from_specs(pkg_specs);
-    context.set_load_system_repo(false);
+    if (resolve_option->get_value() && !alldeps_option->get_value()) {
+        context.set_load_system_repo(true);
+    } else {
+        context.set_load_system_repo(false);
+    }
     context.set_load_available_repos(Context::LoadAvailableRepos::ENABLED);
 }
 
 void DownloadCommand::run() {
     auto & ctx = get_context();
+    auto settings = libdnf::GoalJobSettings();
```

Comment:
* Please move this line to the condition `if (resolve_option->get_value()) {`. It is not used elsewhere.
* C++ is very flexible. Here we are creating an instance of an object on the stack. We can replace this line with `libdnf::GoalJobSettings settings;`. Which I prefer.

An example with a `std::string` object: `auto s = std::string();` can be written as `auto s = std::string{};` and can be replaced with `std::string s;`.

Example with primitive type `int`: `auto a = int();` can be written as `auto a = int{};` and is the same as `int a{};`.